index.js

```diff
@@ -93,69 +84,70 @@ async function ensureRepoIsAccessible(repo) {
 }
 
 async function init() {
-	await verifyToken();
-	const query = new URLSearchParams(location.search);
-	let match;
+	await waitForToken();
+
+	let repo;
+	let ref;
+	let dir;
+
 	try {
+		const query = new URLSearchParams(location.search);
 		const parsedUrl = new URL(query.get('url'));
-		match = repoDirRegex.exec(parsedUrl.pathname);
-		if (!match) {
-			return updateStatus();
-		}
+		[, repo, ref, dir] = repoDirRegex.exec(parsedUrl.pathname);
+
+		console.log('Source:', {repo, ref, dir});
 	} catch (_) {
 		return updateStatus();
 	}
 
-	const [, repo, ref, dir] = match;
-
-	console.log('Source:', {repo, ref, dir});
-
 	if (!navigator.onLine) {
 		updateStatus('⚠ You are offline.');
 		throw new Error('You are offline');
 	}
 
 	updateStatus('Retrieving directory info…');
 
-	await ensureRepoIsAccessible(repo);
+	await validateInput(repo);
```

Comment:
```suggestion
	await validateProvidedRepo(repo);
```

Not married to this exact suggestion, but I think `validateInput` is a little bit too broad / not self-explaining enough.

Reply:
The function validates both the token and the url, which is the entirety of the input

Reply:
You're right. Got myself distracted by the now renamed `verifyToken` function and thought they were distinct validation functions. 🤦‍♂

Reply:
Yeah `verifyToken` itself was ambiguous too